lib/Wikia/src/Consul/Client.php

```diff
@@ -57,31 +60,128 @@ function( $item ) {
 	 *
 	 * @param string $hostname
 	 * @return array list of IP addresses
+	 * @throws \Exception
 	 */
-	function getNodesFromHostname( $hostname ) {
+	function getNodesFromHostname( string $hostname ) {
 		Assert::true( self::isConsulAddress( $hostname ), __METHOD__ . ' should get a Consul address', $this->getLoggerContext() );
 
-		list( $tag, $service ) = explode( '.', $hostname );
-		return $this->getNodes( $service, $tag );
+		if ( self::isConsulQuery($hostname) ) {
+			// e.g. geo-db-g-slave.query.consul
+			$query = self::parseConsulQuery( $hostname );
+			return $this->getNodesFromConsulQuery( $query );
+		}
+		elseif ( self::isConsulServiceAddress( $hostname ) ) {
+			// e.g. slave.db-g.service.consul
+			list( $tag, $service ) = self::parseConsulServiceAddress( $hostname);
+			return $this->getNodes( $service, $tag );
+		}
+		else {
+			throw new \Exception( __METHOD__ . " - {$hostname} is neither consul query nor consul service address" );
+		}
 	}
 
 	/**
-	 * Example: Wikia\Consul\Catalog::isConsulAddress( 'slave.db-smw.service.consul' )
+	 * Example: Wikia\Consul\Client::isConsulAddress( 'slave.db-smw.service.consul' )
 	 *
-	 * @param $address
+	 * @param string $address
 	 * @return bool true if the given address is a consul one
 	 */
-	static function isConsulAddress( $address ) {
+	static function isConsulAddress( string $address ) : bool {
 		return endsWith( $address, '.consul' );
 	}
 
 	/**
+	 * Example: Wikia\Consul\Client::isConsulServiceAddress( 'slave.db-smw.service.consul' )
+	 *
+	 * @param string $address
+	 * @return bool true if the given address is a consul service one
+	 */
+	static function isConsulServiceAddress( string $address ) : bool {
+		return endsWith( $address, '.service.consul' );
+	}
+
+	/**
+	 * Returns a service and tag part from consul address.
+	 *
+	 * Example: 'slave.db-smw.service.consul' => ['slave', 'db-smw']
+	 *
+	 * @param string $address
+	 * @return string[]|false
+	 */
+	static function parseConsulServiceAddress( string $address ) {
+		if ( self::isConsulServiceAddress( $address ) ) {
+			list( $tag, $service ) = explode( '.', $address, 3 );
+			return [ $tag, $service ];
+		}
+		else {
+			return false;
+		}
+	}
+
+	/**
+	 * Example: Wikia\Consul\Client::isConsulQuery( 'geo-db-g-slave.query.consul' )
+	 *
+	 * @see https://www.consul.io/api/query.html
+	 *
+	 * @param string $address
+	 * @return bool true if the given address is a consul query
+	 */
+	static function isConsulQuery( string $address ) : bool {
+		return endsWith( $address, '.query.consul' );
+	}
+
+	/**
+	 * Returns a query from consul address.
+	 *
+	 * It basically returns the first part of the dot-separated domain
+	 *
+	 * Example: 'geo-db-g-slave.query.consul' => 'geo-db-g-slave'
+	 *
+	 * @param string $address
+	 * @return string|false
+	 */
+	static function parseConsulQuery( string $address ) {
+		return self::isConsulQuery( $address )
+			? strtok( $address, '.' )
+			: false;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the literal "false" boolean value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)

[project-id]: wikia:app